### PR TITLE
fix: fixed incorrect error when max grade 100

### PR DESCRIPTION
**Pull request by martinwang-d2l**
Trello: https://trello.com/c/wUChAmvc/516-subviews-creating-additional-view-on-new-quiz-with-specified-grade-range-triggers-input-validation-incorrectly

Issue cause:

- When making changes to min grade or max grade, grade saved as string instead of numerical value. When creating a new view, comparing min, max compares two strings, which is why we get '30' being larger than '100', causing the error to incorrectly appear. 
- Issue was not present with editing an existing view, because existing value loaded in from siren-sdk as a number. 

Fixed by casting to int during comparison.

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js [hunk 1]
```diff
@@ -211,7 +211,7 @@ class ActivityQuizSubmissionViewsAccessRestrictionsEditor
 	}
 
 	_isInputGradeValid(value) {
-		const targetValue = (value === undefined) ? null : String(value);
+		const targetValue = (value === undefined) ? null : value;
```
Comment (dustxd) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js:214]: Is there a reason why we were string casting it before?

**Review by dustxd — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js (hunk 1 above)
Comment (martinwang-d2l) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js:214]: I think this was from an earlier attempt to patch a null value, it's not used anymore though

Comment (dustxd) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-access-restrictions-editor.js]: was that for the case where either min or max doesn't contain a value?

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-view-release-description.js [hunk 2]
```diff
@@ -68,6 +68,10 @@ class ActivityQuizSubmissionViewReleaseDescription
 			maxGrade = (gradeRestrictionsMaxGrade && gradeRestrictionsMaxGrade.value) || 0;
 		}
 
+		if (maxGrade === 100) {
+			maxGrade = 0;
```
Comment (dustxd) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-view-release-description.js:72]: Why are we setting this to 0?

**Review by dustxd — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-dialog-card-editor.js [hunk 3]
```diff
@@ -73,7 +73,7 @@ class ActivityQuizSubmissionViewsDialogCardEditor
 		const percentComponents = Array.from(accessRestrictions.shadowRoot.querySelectorAll('d2l-input-percent'));
 
 		const existGrades = gradeRestrictionsMinGrade.value && gradeRestrictionsMaxGrade.value;
-		const gradeRestrictionsValid = (existGrades) ? gradeRestrictionsMinGrade.value < gradeRestrictionsMaxGrade.value : true;
+		const gradeRestrictionsValid = (existGrades) ? parseInt(gradeRestrictionsMinGrade.value) < parseInt(gradeRestrictionsMaxGrade.value) : true;
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-submission-views-dialog-card-editor.js:76]: What happens when one of the values is in fact null at this point?

**Review by mdulat — APPROVED** [review]

**Pull request merged**